### ISSUE TITLE
Can't install component 'i686-pc-windows-gnu' on linux

### ISSUE THREAD

**Issue by Thinkofname** [rust-lang/rustup#1127]
```
info: downloading component 'rust-std' for 'i686-pc-windows-gnu'
info: installing component 'rust-std' for 'i686-pc-windows-gnu'
info: rolling back changes
error: could not copy file from '/root/.rustup/tmp/b1vtfeiidqc5ro74_dir/rust-std-i686-pc-windows-gnu/lib\rustlib\i686-pc-windows-gnu\lib\arena-04cd199c7c4cce31.dll' to '/root/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib\rustlib\i686-pc-windows-gnu\lib\arena-04cd199c7c4cce31.dll'
info: caused by: the source path is not an existing regular file
```
From the looks of the error is seems Windows style paths seem to be being used instead of linux style

**Comment by Thinkofname** [comment]
Looking into it this seems to be an issue with rust-installer. 

https://github.com/rust-lang/rust-installer/blob/daa2a05ebe7b8d07a309e8891ebc548652362954/src/generator.rs#L111-L129
Its using `display` which is placing platform specific paths into the manifest.in

I'm not sure which side should be fixed however

As a workaround something like this would work but i'm not sure if there is a better way?
```diff
diff --git a/src/rustup-dist/src/component/package.rs b/src/rustup-dist/src/component/package.rs
index e03b8d2..b6296bf 100644
--- a/src/rustup-dist/src/component/package.rs
+++ b/src/rustup-dist/src/component/package.rs
@@ -94,7 +94,7 @@ impl Package for DirectoryPackage {
             let part = try!(ComponentPart::decode(l)
                             .ok_or_else(|| ErrorKind::CorruptComponent(name.to_owned())));
 
-            let path = part.1;
+            let path: PathBuf = part.1.to_string_lossy().replace('\\', "/").into();
             let src_path = root.join(&path);
 
             match &*part.0 {
```

**Comment by Diggsey** [comment]
I reproduced this, but it only happens on nightly. Something must have changed in how the components are built and packaged, which makes this an upstream issue.

**Comment by Diggsey** [comment]
Reported as rust-lang/rust-installer#65

**Comment by onur** [comment]
If you really need to install this target (x86_64-pc-windows in this case) until this issue resolved, you can find downloaded tarball in `~/.rustup/downloads`, manually extract it with `tar xf` and move `rust-std-nightly-x86_64-pc-windows-gnu/rust-std-x86_64-pc-windows-gnu/lib/rustlib/x86_64-pc-windows-gnu/` directory into `~/.rustup/toolchains/nightly-{YOUR_NIGHTLY_VERSION}-x86_64-unknown-linux-gnu/lib/rustlib/`.

**Comment by cuviper** [comment]
The manifests should be fixed already by rust-lang/rust#42132.

**Comment by Thinkofname** [comment]
Yep, fixed here for me
